LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -111,7 +125,7 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
 
             try
             {
-                var bundlerResult = await TryUseBundler(request, loRaDevice, loraPayload, useMultipleGateways);
+                var bundlerResult = await TryUseBundler(request, loRaDevice, loraPayload, useMultipleGateways, concentratorDeduplicationResult);
```

Comment:
can't add a comment on that line, as it wasn't changed (thanks gitHub), but on line 147 we do Perform the ADR not matter if it was a duplicate or not. Meaning that we have a roundtrip to the server no matter if it is a duplicate or not. 

We discussed this, and this might be desired. The ADR is a bit tricky to get right with multi concentrators, as the values will differ from each concentrator. If we collect all values from all the concentrators, we may adapt invalid parameters that satisfy all concentrators on avg, failing to optimize the DR to what it would be possible. 

A simple option would be to skip the ADR collection for duplicate or soft duplicate messages and only collect for the first one. We would make the assumption that the first one getting the message would be the one closest to the device (likely, but not necessary). In theory, the ADR would have to be adapted to take multiple concentrators into consideration and optimize for the one closest to the device.

Reply:
One more thing: if we do the ADR for all of them, the requiresConfirmation may be reset and we end up sending multiple confirmations for that message (if the ADR table is complete). 

I suggest to do the following:

1. disable ADR processing for duplicate concentrator messages
2. add a backlog item to look at the ADR implementation and take multi concentrators into consideration

/cc @Mandur  to chime in here

Reply:
Thanks for raising this, we said on our discussions last week to put ADR as out of scope but it was a bit mentioned on the side and we didn't define what "out of scope" means exactly. So it's good to discuss this, do you think we need to involve Ronnie for this decision? 

> disable ADR processing for duplicate concentrator messages

Here you mean the cases Duplicate, SoftDuplicate and Resubmission?

Reply:
- Added a check to skip for soft duplicates in 7cb1b563072321e31e9cd8581cb87feeb46f78f6.
- Added a backlog item to re-enable ADR for multi concentrator support #1017 that could be discussed with Ronnie or @Mandur

Reply:
Added also a specific integration test that will break if we enable ADR for Soft duplicates: `When_ADR_Enabled_PerformADR_Should_Be_Disabled_For_SoftDuplicate`